lmcache/cache_engine.py

```diff
@@ -301,7 +316,7 @@ def retrive(self,
         for chunk in retrival_iterator:
             if chunk is None:
                 break
-            retrived_kv_chunks.append(chunk)
+            retrived_kv_chunks.append(chunk.to(device))
```

Comment:
Why do `to(device)` here rather than after the concatenation? It is a bit counter-intuitive that this will be faster.

Reply:
It is 2X faster because the cat operation is performed on gpu.